projects/core/src/cart/facade/active-cart.service.ts

```diff
@@ -409,7 +439,11 @@ export class ActiveCartService implements OnDestroy {
    * @param email
    */
   addEmail(email: string): void {
-    this.multiCartService.assignEmail(this.cartId, this.userId, email);
+    this.activeCartId$
+      .pipe(withLatestFrom(this.userIdService.getUserId()), take(1))
```

Comment:
```
    this.activeCartId$
      .pipe(withLatestFrom(this.userIdService.getUserId()), take(1))
```
maybe an opportunity to move it to some private function, like `grabActiveCartAndUserId`, but just sharing a thought.